### PR TITLE
Mccalluc/collections citation

### LINKED ISSUE #1709: Implement Citation section on Collections page
See design in #1578

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #1709 
- @tsliaw : Do you know if 6 is the precise cut-off for the authors list? The mockup has a info-button beside the header... is it necessary, and what should it say?
- @john-conroy : code review.

### CODE REVIEW

File: context/app/static/js/components/Detail/Citation/Citation.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    .map(({ last_name, first_name }) => `${last_name} ${first_name[0]}`);
+  if (firstContributors.length < contributors.length) {
+    firstContributors.push('et al');
+  }
+  const year = new Date(timestamp).getFullYear();
+  return (
+    <SectionContainer id="citation">
+      <SectionHeader>Citation</SectionHeader>
+      <FlexPaper>
+        {firstContributors.join(', ')}. {title} [Internet]. HuBMAP Consortium; {year}. Available from:{' '}
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Citation/Citation.jsx:21]: Perhaps use a template string?

File: context/app/static/js/components/Detail/Citation/Citation.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { LightBlueLink } from 'js/shared-styles/Links';
+import SectionHeader from 'js/shared-styles/sections/SectionHeader';
+import SectionContainer from 'js/shared-styles/sections/SectionContainer';
+import { FlexPaper } from './style';
+
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Citation/Citation.jsx:8]: Include a comment which includes the citation format we're using?

File: context/app/static/js/pages/Collection/Collection.jsx [hunk 3]
```diff
@@ -47,6 +48,7 @@ function Collection(props) {
               </OutboundLink>
             )}
           </Summary>
+          <Citation contributors={creators} title={title} timestamp={create_timestamp} url={doi_url} />
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply context/app/static/js/pages/Collection/Collection.jsx:51]: It should be on all collections: It's a piece of core metadata for DOI creation, which is being done for each of these.

File: context/app/static/js/components/Detail/Citation/style.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import styled from 'styled-components';
+import Paper from '@material-ui/core/Paper';
+
+const FlexPaper = styled(Paper)`
+  padding: 30px 40px 30px 40px;
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply context/app/static/js/components/Detail/Citation/style.js:5]: There are lots of these across the codebase... maybe factor it out more broadly? Will file an issue. #1808

File: context/app/static/js/components/Detail/Citation/Citation.jsx (hunk 1 above)
Comment (mccalluc) [reply context/app/static/js/components/Detail/Citation/Citation.jsx]: Thanks for the nudge: factored this out to a function and added unit test.

**Comment by tsliaw** [comment]
> Do you know if 6 is the precise cut-off for the authors list?

NLM formatting should be a list of all authors, I'll fix that in the mockup. 

> The mockup has a info-button beside the header... is it necessary, and what should it say?

Screen Shot 2021-05-05 at 10 50 27 AM
This is a screenshot of the mockup that has what the info button should say. 

Can you increase the font size to 'body1' for the citation text? I think it's currently 'body2' in this PR.

File: context/app/static/js/components/Detail/Citation/Citation.jsx (hunk 2 above)
Comment (mccalluc) [reply context/app/static/js/components/Detail/Citation/Citation.jsx:9]: The function name now has "NLM".

**Pull request merged**